src/spatialite.mk

```diff
@@ -0,0 +1,44 @@
+# This file is part of MXE. See LICENSE.md for licensing information.
+
+PKG             := spatialite
+$(PKG)_WEBSITE  := https://www.gaia-gis.it/fossil/libspatialite/index
+$(PKG)_DESCR    := SpatiaLite
+$(PKG)_IGNORE   :=
+$(PKG)_VERSION  := 4.4.0-RC1
+$(PKG)_CHECKSUM := 80f7fff0a147044c5eb197e565f598ac1f137d86d0a548cbc8f52fb7ff7cac68
+$(PKG)_SUBDIR   := libspatialite-$($(PKG)_VERSION)
+$(PKG)_FILE     := libspatialite-$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := http://www.gaia-gis.it/gaia-sins/libspatialite-sources/$($(PKG)_FILE)
+$(PKG)_DEPS     := dlfcn-win32 freexl geos sqlite zlib proj libiconv libxml2
+
+define $(PKG)_UPDATE
+    $(WGET) -q -O- 'http://www.gaia-gis.it/gaia-sins/libspatialite-sources/' | \
+    $(SED) -n 's,.*libspatialite-\([0-9][^>]*\)\.tar.*,\1,p' | \
+    tail -1
+endef
+
+define $(PKG)_BUILD
+    # freeXL support is only necessary if you want to be able to parse .xls files.
+    # If you disable freexl support, remove -lfreexl from the test program below.
+    cd '$(1)' && autoreconf -fi -I ./m4
+    cd '$(1)' && ./configure \
+    $(MXE_CONFIGURE_OPTS) \
+    --enable-freexl=yes \
+    --with-geosconfig='$(PREFIX)/$(TARGET)/bin/geos-config'
+    $(MAKE) -C '$(1)' -j '$(JOBS)' $(if $(BUILD_SHARED), LDFLAGS='-no-undefined')
+    $(MAKE) -C '$(1)' -j 1  $(INSTALL_STRIP_LIB)
+
+    # compile one of the demo programs (copied from the source package)
```

Comment:
Could build test directly from source instead of copying to MXE tree.

Reply:
I had considered this to a degree - the test in the MXE tree is subtly edited so that it will compile with the -Werror flag set (as far as I recall, there are pointer casting warnings that prevent the test from compiling.)
Should I relax the test's compilation flags to be able to build the test directly from the source path?

Reply:
Our tests are mostly to confirm that linking works and serve as an example of how one might use whatever configuration system the package provides. Strict flags are appropriate for hand-written tests; they can be relaxed when using existing sources.

The runtime side of the test can also be relaxed, creating batch files to set env vars (for example) is a good thing but trying to exercise the functionality of a package is mostly out of our scope.

Our time is better spent on improving the configuration for downstream users (via pkg-config, cmake, et al) so they can incorporate packages with minimal manual specification of libs, flags, etc.